compiler/rustc_index/src/slice.rs

```diff
@@ -20,7 +20,7 @@ pub struct IndexSlice<I: Idx, T> {
 
 impl<I: Idx, T> IndexSlice<I, T> {
     #[inline]
-    pub const fn empty() -> &'static Self {
+    pub const fn empty<'a>() -> &'a Self {
```

Comment:
Is this because of invariance or what?

Reply:
Yeah this function is used here https://github.com/rust-lang/rust/blob/4edbd9320b39592a2af26ca5cb68018fadebc3fa/compiler/rustc_ty_utils/src/layout.rs#L374-L380

And changing the lifetime from `'_` to `'tcx` here https://github.com/rust-lang/rust/blob/4edbd9320b39592a2af26ca5cb68018fadebc3fa/compiler/rustc_ty_utils/src/layout.rs#L119

means that we would now create a `&'static &'tcx Something` which would require `'tcx: 'static`,

Reply:
Ah, ok. Yeah, that makes sense 👍